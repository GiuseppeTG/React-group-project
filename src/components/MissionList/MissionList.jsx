import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { getMissions } from '../../redux/Missions/Missions';
import './MissionList.scss';

function MissionsList() {
  const missions = useSelector((state) => state.missions.missions);
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(getMissions());
  }, []);
  return (
    <div className="missions-container">
      <h2 className="missions-title">MISSIONS</h2>
      <div className="missions-list">
        {missions.map((mission) => (
          <div key={mission.id} className="mission-container">
            <div className="mission-header">
              <h3 className="mission-name">{mission.name}</h3>
              <h3 className="mission-status">
                {mission.reserved ? 'ACTIVE MEMBER' : 'NOT A MEMBER'}
              </h3>
            </div>
            <p className="mission-description">{mission.description}</p>
          </div>
        ))}
        ;
      </div>
    </div>
  );
}

export default MissionsList;
