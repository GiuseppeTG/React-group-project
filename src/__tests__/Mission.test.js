import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import MissionList from '../components/MissionList/MissionList';
import store from '../redux/configureStore';

describe('Testing component rednering propperly', () => {
  test('renders MissionsList component', () => {
    const tree = render(
      <BrowserRouter>
        <Provider store={store}>
          <MissionList />
        </Provider>
      </BrowserRouter>,
    );
    expect(tree).toMatchSnapshot();
  });
});
