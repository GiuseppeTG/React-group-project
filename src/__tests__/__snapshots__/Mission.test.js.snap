// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing component rednering propperly renders MissionsList component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    data-aos-delay="0"
    data-aos-duration="2000"
    data-aos-easing="ease"
  >
    <div>
      <div
        class="missions-container"
        data-testid="missions-list-test"
      >
        <h2
          class="missions-title"
        >
          MISSIONS
        </h2>
        <div
          class="missions-list"
        >
          ;
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="missions-container"
      data-testid="missions-list-test"
    >
      <h2
        class="missions-title"
      >
        MISSIONS
      </h2>
      <div
        class="missions-list"
      >
        ;
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
